### PR TITLE
Use nvidia-container-toolkit in ml-test scripts

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
Replace nvidia-docker2 with nvidia-container-toolkit in ml-test scripts as the former is [deprecated](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/release-notes.html#nvidia-container-toolkit-1-14-0).
After nvidia-docker2 version upgraded to 2.14.0, we started seeing the following error:
   ```
   docker: Error response from daemon: unknown or invalid runtime name: nvidia.
   See 'docker run --help'.
   ```

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Ran the tests manually. The error is no longer there.
> Metrics:
>
> network throughput ~ 25 MiB/sec
> CPU usage ~ 59%
> Memory usage ~ 13-18%
>
>Time in the epoch logs is in similar range before and after upgrade (~ 0.700000)
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/ml_tests/setup_host.sh [hunk 1]
```diff
@@ -38,12 +38,11 @@ sudo curl -fSsl -O $BASE_URL/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION
...
-            sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
-            sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
+  && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
+    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
+    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
 
 sudo apt-get update
-sudo apt-get install -y nvidia-docker2
+sudo apt-get install -y nvidia-container-toolkit
```
Comment (raj-prince) [inline perfmetrics/scripts/ml_tests/setup_host.sh:47]: Do we also need to configure docker for runtime? Or could you please check the content of `/etc/docker/daemon.json` file?

Ref: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#configuring-docker

File: perfmetrics/scripts/ml_tests/setup_host.sh [hunk 2]
```diff
@@ -38,12 +38,11 @@ sudo curl -fSsl -O $BASE_URL/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION
...
-      && curl -s -L https://nvidia.github.io/libnvidia-container/$distribution/libnvidia-container.list | \
-            sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
-            sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
+  && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
+    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
+    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
 
 sudo apt-get update
-sudo apt-get install -y nvidia-docker2
```
Comment (raj-prince) [inline perfmetrics/scripts/ml_tests/setup_host.sh:48]: Just to confirm, this issue will not come if we install nvidia-docker2 (v2.13.0)?

Comment (ashmeenkaur) [reply perfmetrics/scripts/ml_tests/setup_host.sh:48]: Yes but we should not since it is deprecated.

**Review by raj-prince — COMMENTED** [review]
LGTM.

Could you please compare the throughput, cpu and memory usage with the previous run?

File: perfmetrics/scripts/ml_tests/setup_host.sh (hunk 1 above)
Comment (ashmeenkaur) [reply perfmetrics/scripts/ml_tests/setup_host.sh:47]: As per this [documentation](https://gitlab.com/nvidia/container-toolkit/container-toolkit/-/tree/main/cmd/nvidia-container-runtime#notes-on-using-the-docker-cli), the legacy nvidia-container-runtime mode is directly compatible with the --gpus flag implemented by the docker CLI so I believe we don't need this.

**Review by raj-prince — APPROVED** [review]

**Pull request merged**